packages/docs/guide/advanced/router-view-slot.md

```diff
@@ -0,0 +1,75 @@
+# RouterView slot
+
+The RouterView component exposes a slot that can be used to render the route component:
+
+```vue-html
+<router-view v-slot="{ Component }"
+  <component :is="Component" />
+</router-view>
+```
+
+The code above is equivalent to using `<router-view />` without the slot, but the slot provides extra flexibility when we want to work with other features.
+
+## Passing props and slots
+
+We could use the slot to pass props or slots to the route component:
+
+```vue-html
+<router-view v-slot="{ Component }"
+  <component :is="Component" some-prop="a value">
+    <p>Some slotted content</p>
+  </component>
+</router-view>
+```
+
+In practice, this usually isn't something you would want to do, as the route components would all need to use the same props and slots. See [Passing Props to Route Components](/guide/essentials/passing-props) for other ways to pass props.
+
+## Template refs
+
+Using the slot also allows us to put a [template ref](https://vuejs.org/guide/essentials/template-refs.html) directly on the route component:
+
+```vue-html
+<router-view v-slot="{ Component }"
+  <component :is="Component" ref="mainContent" />
+</router-view>
+```
+
+If we put the ref on the `<router-view>` instead then the ref would be populated with the RouterView instance, rather than the route component.
+
+## KeepAlive
```

Comment:
let's move this up to be the first one and add an example that also uses transition at the same time. I think it's the main use case for the slot

Reply:
This should be done. Let me know if this isn't what you had in mind.